### PR TITLE
Add function write_sessioninfo()

### PR DESCRIPTION
**Pull request by florisvdh**
The function writes essentials of `sessioninfo::session_info()` to a file or connection in order to maintain versionable documentation of a workflow and aid reproducibility.

### CODE REVIEW

**Review by peterdesmet — COMMENTED** [review]
Sorry, no time to review this one.

File: R/write_sessioninfo.R [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#' capture.output
+#' @importFrom rlang
+#' .data
+#' @importFrom dplyr
+#' mutate
+#' select
+#'
+#' @export
+#'
+#' @examples
```
Comment (hansvancalster) [inline R/write_sessioninfo.R:34]: I've added this example. @florisvdh How do you see the use of this function in a workflow? I have something like this in mind:

- calculate a hash on the created txt file with session info
- check the hash to make sure nothing changed

?

File: R/write_sessioninfo.R [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+#' \dontrun{
+#' write_sessioninfo(file = "./sessioninfo.txt")
+#' }
+write_sessioninfo <- function(file) {
```
Comment (hansvancalster) [inline R/write_sessioninfo.R:38]: The function writes both platform info (tab separated) and package info (space separated) to the same text file. Whether this is OK depends a bit on the workflow you have in mind. An alternative might be to return package info as a data.frame (or file) with platform info as attributes or as a separate data.frame (or file)?

File: R/write_sessioninfo.R (hunk 1 above)
Comment (florisvdh) [reply R/write_sessioninfo.R:34]: The main purpose is to document the environment in which a versioned workflow (R script, Rmd) is being executed. Beside the pure documentation & traceability functionality, it eases troubleshooting when another user wants to reproduce the outcome (e.g. did I get an error because I use another package version?).

Versioning either the txt file or a hash file would make visible whether changes have occurred, but only versioning of the txt file will make clear which changes have occurred. The advantage of a hash is that it is smaller (that is the strategy used in git LFS), but I think this advantage is rather marginal here.

So the implementation (and my recommendation) goes like: use `write_sessioninfo()` at the end of each workflow in order to also document (and version) your environment, not just your code.

File: R/write_sessioninfo.R (hunk 2 above)
Comment (florisvdh) [reply R/write_sessioninfo.R:38]: I just wanted a simple text file that holds this info in a specific format and that can be versioned (typically with git). The dataframe approach generally needs R to inspect dataframe contents and differences (but depending on how you store it, you could as well inspect the associated file directly: csv, tsv, ...). In this case I see the dataframe format as an unneeded complexity - do you want to load it as dataframe back into R?

It is true that the precise formatting of the file is rather arbitrary, alternatives may work just as well.

File: R/write_sessioninfo.R [hunk 3]
```diff
@@ -28,7 +28,13 @@
 #' @importFrom dplyr
 #' mutate
 #' select
+#'
 #' @export
+#'
+#' @examples
+#' \dontrun{
+#' write_sessioninfo(file = "./sessioninfo.txt")
```
Comment (florisvdh) [inline R/write_sessioninfo.R]: @hansvancalster I suggest to simplify your example as `write_sessioninfo(file = "sessioninfo.txt")`

File: inborutils.Rproj [hunk 4]
```diff
@@ -18,3 +18,5 @@ StripTrailingWhitespace: Yes
 BuildType: Package
 PackageUseDevtools: Yes
 PackageInstallArgs: --no-multiarch --with-keep.source
+PackageCheckArgs: --as-cran
+PackageRoxygenize: rd,collate,namespace
```
Comment (florisvdh) [inline inborutils.Rproj:22]: @hansvancalster I don't know the precise implications of these two lines, but it is package wide and should probably  be reviewed in a separate PR?

File: R/write_sessioninfo.R (hunk 1 above)
Comment (hansvancalster) [reply R/write_sessioninfo.R:34]: I see, I made it too complex. That's OK.

File: R/write_sessioninfo.R (hunk 2 above)
Comment (hansvancalster) [reply R/write_sessioninfo.R:38]: I agree with your take on it.

File: inborutils.Rproj (hunk 4 above)
Comment (hansvancalster) [reply inborutils.Rproj:22]: Sorry :wink: I just misused your PR to add these lines to the .Rproj file. They aid in checking the package when it is built. @ThierryO advised to include these settings in the .Rproj file and he can tell you more.

File: R/write_sessioninfo.R (hunk 3 above)
Comment (hansvancalster) [reply R/write_sessioninfo.R]: Done in 4c0dc7a

**Review by hansvancalster — APPROVED** [review]

File: inborutils.Rproj (hunk 4 above)
Comment (florisvdh) [reply inborutils.Rproj:22]: You're welcome.

**Comment by florisvdh** [comment]
@hansvancalster Thanks for your reflections and adding an example!

**Comment by ThierryO** [comment]
I suggest to store the information in YAML format. See #64

**Comment by hansvancalster** [comment]
@florisvdh I think this PR is superseded by #64? Can you close this one if my assumption is correct?

**Comment by hansvancalster** [comment]
Something to discuss with @inbo/bmk

**Comment by florisvdh** [comment]
This is an old PR it seems. Nowadays, there is:

`sessioninfo::session_info(to_file = TRUE)`

and there are more sophisticated possibilities with {renv} and a few other packages I think.

Hence I think you can close this PR @hansvancalster.
